backend/graphql/resolvers/courseResolvers.ts

```diff
@@ -16,14 +20,70 @@ import UserService from "../../services/implementations/userService";
 import IEmailService from "../../services/interfaces/emailService";
 import IUserService from "../../services/interfaces/userService";
 import { Role } from "../../types";
-import { CourseVisibilityAttributes } from "../../models/course.model";
+import { CourseVisibilityAttributes, Module } from "../../models/course.model";
 import { assertNever } from "../../utilities/errorUtils";
 
 const courseService: ICourseService = new CourseService();
 const userService: IUserService = new UserService();
 const emailService: IEmailService = new EmailService(nodemailerConfig);
 const authService: IAuthService = new AuthService(userService, emailService);
 
+function serializeToModuleDTO(modules: { [id: string]: Module }): ModuleDTO[] {
+  return Object.entries(modules).map(([key, val]) => ({ id: key, ...val }));
+}
+
+function deserializeModuleDTO(
+  moduleDTOs: ModuleDTO[],
+): { [id: string]: Module } {
+  const modules: { [id: string]: Module } = {} as { [id: string]: Module };
+
+  moduleDTOs.forEach((moduleDTO) => {
+    const moduleObj = {
+      title: moduleDTO.title,
+      description: moduleDTO.description,
+      image: moduleDTO.image,
+      previewImage: moduleDTO.previewImage,
+      published: moduleDTO.published,
+      lessons: moduleDTO.lessons,
+    };
+
+    // If no UUID is defined, create one
+    modules[moduleDTO?.id ?? uuidv4()] = moduleObj;
+  });
+
+  return modules;
+}
+
+function serializeCourseResponse(
+  course: CourseResponseDTO,
+): SerializedCourseResponseDTO {
+  if (course?.modules) {
+    return {
+      ...course,
+      modules: serializeToModuleDTO(course.modules),
+    };
+  }
+  return {
+    ...course,
+    modules: [],
+  };
```

Comment:
- Doesn't `course` always exist? Why the `?` operator

Would this work?
```suggestion
  return {
    ...course,
    modules: course.modules ? serializeToModuleDTO(course.modules) : [],
  };
```

Reply:
Also, shouldn't `modules` always exist? It might just be empty (I don't remember how it's implemented right now, but initially the `lessons` parameter was required on create course)

Reply:
I wasn't sure, iirc lessons was optional in the graphql type previously so I made modules optional as well, and I think it makes sense to have a newly created course with no modules. Although now that I think about it, it makes more sense to remove this handling and have the frontend always send `[]` to indicate no modules.